lib/flow.rb

```diff
@@ -1,14 +1,28 @@
+require 'pry'
+
 class Flow
 
-  def initialize(jurisdiction, type)
-    path = get_path(jurisdiction, type)
+  # will generate a flow chart based on the path that can be navigated through the
+  # entire scope of the certificates (legal and practical)
+  attr_accessor :xml, :xml_copy
+
+  def initialize(jurisdiction, type, path_param = false)
+    if path_param.present?
+      path = path_param
+    else
+      path = get_path(jurisdiction, type)
+    end
+    # two Nokogiri parses are created because the operations & transformations exacted destructively modify data
     @xml = Nokogiri::XML(File.open(path))
     @xml_copy = Nokogiri::XML(File.open(path))
+    # marshals XML into two above hashes
+
   end
 
   def get_path(jurisdiction, type)
+    # return two different paths based on whether legal or practical question flow selected
     jurisdiction = "en" if (jurisdiction == "gb" && type == "Practical")
-    jurisdiction.upcase! if type == "Legal"
+    jurisdiction.upcase! if type == "Legal" # check what this bang! means
```

Comment:
This performs the transformation in place, so we don't have to do jurisdiction = jurisdiction.upcase